test/shopping-list.test.js

```diff
@@ -0,0 +1,162 @@
+    it("should be a function", function () {
+      expect(steak.render).to.be.a("function");
+    })
+
+    it("should return a string", function () {
+      expect(steak.render()).to.be.a("string");
+
+    })
+
+    it("should start with an <li tag", function () {
+      expect(steak.render().slice(0, 3)).to.equal('<li');
+
+    })
+    it("should end with an </li> tag", function () {
+      expect(steak.render().slice(-5)).to.equal('</li>');
+    })
+  })
+})
+
+describe('Shopping list', function () {
+
+  describe("List is a class", function () {
+    it("should be function", function () {
+      expect(List).to.be.a("function");
+    })
+  })
+  describe("has a property named 'items'", function () {
+    let shoppingList = new List();
+    it("should be function", function () {
+      expect(shoppingList).to.have.property("items");
+    })
+    it('the items property should be an array', function () {
+      expect(shoppingList.items).to.be.a("array");
+    })
+  })
+  describe("has a method named addItem", function () {
+    let shoppingList = new List();
+    it("should be a function", function () {
+      expect(shoppingList.addItem).to.be.a("function");
+    })
+    it("should add an Item object to the items array", function () {
+      let steak = new Item('steak', 't-bone steak');
+      shoppingList.addItem(steak);
+      expect(shoppingList.items[0]).to.deep.equal(steak);
+    })
+    it("should return an error with anything other than an Item object", function () {
+      expect(shoppingList.addItem.bind(shoppingList, 'cookies')).to.throw();
+    })
+  })
+  describe("has a method named removeItem", function () {
+    let shoppingList = new List();
+    it("should be a function", function () {
+      expect(shoppingList.removeItem).to.be.a("function");
+    })
+    it("should remove an Item object from the items array", function () {
+      let steak = new Item('steak', 't-bone steak');
+      let cookies = new Item('cookies', 'chocolate chip');
+      shoppingList.items.length = 0;
+      shoppingList.addItem(steak);
+      shoppingList.addItem(cookies);
+      shoppingList.removeItem(steak);
+      expect(shoppingList.items[1]).to.equal(undefined);
+    })
+    it("should return an error if item does not exist in the list", function () {
+      let steak = new Item('steak', 't-bone steak');
+      let cookies = new Item('cookies', 'chocolate chip');
+      shoppingList.items.length = 0;
+      shoppingList.addItem(cookies);
+      expect(shoppingList.removeItem.bind(shoppingList, steak)).to.throw();
+    })
+    it("should remove the last item in the list if there is no input", function () {
+      let steak = new Item('steak', 't-bone steak');
+      let cookies = new Item('cookies', 'chocolate chip');
+      shoppingList.items.length = 0;
+      shoppingList.addItem(cookies);
+      shoppingList.addItem(steak);
+      shoppingList.removeItem();
+      expect(shoppingList.items[1]).to.equal(undefined);
```

Comment:
This should also test to see that there's only one item in the array and that the item in the array is a cookie.